components/org.wso2.carbon.status.dashboard.core/src/main/java/org/wso2/carbon/status/dashboard/core/impl/WorkersApiServiceImpl.java

```diff
@@ -1281,12 +1309,47 @@ public Response getWorkerConfig(String id, String username) throws NotFoundExcep
      * @throws NotFoundException
      */
     @Override
-    public Response testConnection(String auth, String username) throws NotFoundException {
+    public Response testConnection(String workerId, String username) throws NotFoundException {
         boolean isAuthorized = permissionProvider.hasPermission(username, new Permission(Constants.PERMISSION_APP_NAME,
                 VIWER_PERMISSION_STRING));
         if (isAuthorized) {
-            return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK,
-                    "This is not supported yet")).build();
+            String[] hostPort = workerId.split(Constants.WORKER_KEY_GENERATOR);
+            int status = 404;
+            if (hostPort.length == 2) {
+                WorkerResponce workerResponce = new WorkerResponce();
+                String uri = generateURLHostPort(hostPort[0], hostPort[1]);
+                try {
+                    feign.Response workerResponse = WorkerServiceFactory.getWorkerHttpsClient(PROTOCOL + uri,
+                            getUsername(),
+                            getPassword()).getWorker();
+                    status = workerResponse.status();
+                    if (status == 200) {
+                        workerResponce.setCode(200);
```

Comment:
spellcheck - reponce

Reply:
Fixed in 84eed0f2066c8da5780a1ab6e5bbed953ff210b9